### PR TITLE
rutube: add support for shorts and yappy

### PR DESCRIPTION
**Pull request by wukko**

### CODE REVIEW

File: src/modules/processing/services/rutube.js [hunk 1]
```diff
@@ -1,18 +1,46 @@
...
-    if ("hls" in play.live_streams) return { error: 'ErrorLiveVideo' };
-    if (!play.video_balancer || play.detail) return { error: 'ErrorEmptyDownload' };
+    if (play.detail || !play.video_balancer) return { error: 'ErrorEmptyDownload' };
+    if (play.live_streams?.hls) return { error: 'ErrorLiveVideo' };
+
+    if (play.duration > maxVideoDuration)
+        return { error: ['ErrorLengthLimit', maxVideoDuration / 60000] };
+
+    let m3u8 = await fetch(play.video_balancer.m3u8)
+                     .then((r) => { return r.text() })
```
Comment (dumbmoron) [inline src/modules/processing/services/rutube.js:41]:
```suggestion
                     .then(r => r.text())
```

Comment (synzr) [reply src/modules/processing/services/rutube.js]: typical wukko code

**Review by dumbmoron — APPROVED** [review]

File: src/modules/processing/services/rutube.js
[the remark this answers is not in the document]
Comment (wukko) [reply src/modules/processing/services/rutube.js:1]: ?

[the remark this answers is not in the document]
Comment (synzr) [reply src/modules/processing/services/rutube.js:1]:
```js
async function requestJSON(url) {
    let r = await fetch(url)
            .then((r) => { return r.json() })
            .catch(() => {});
    return r
    try {
        const r = await fetch(url);
        return await r.json();
    } catch {
        // fuck you
    }
}
```

[the remark this answers is not in the document]
Comment (wukko) [reply src/modules/processing/services/rutube.js:1]: why

[the remark this answers is not in the document]
Comment (synzr) [reply src/modules/processing/services/rutube.js:1]: its funny

**Pull request merged**